File: 백준/Silver/17219. 비밀번호 찾기/비밀번호 찾기.py
import sys
input = sys.stdin.readline

N, M = map(int, input().split())
addrPw = {}
for _ in range(N):
    addr, pw = map(str, input().split())
    addrPw[addr] = pw

result = ""
for _ in range(M):
    request = input().strip()
    result += addrPw[request]+"\n"
print(result, end="")

File: 백준/Silver/17219. 비밀번호 찾기/비밀번호 찾기.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        String[] line = br.readLine().split(" ");
        int N = Integer.parseInt(line[0]);
        int M = Integer.parseInt(line[1]);
        HashMap<String, String> addrPw = new HashMap<>();
        for (int i = 0; i < N; i++) {
            line = br.readLine().split(" ");
            addrPw.put(line[0], line[1]);
        }
        StringBuilder sb = new StringBuilder();
        for (int i = 0; i < M; i++) {
            String request = br.readLine().trim();
            sb.append(addrPw.get(request) + "\n");
        }
        br.close();
        System.out.println(sb.toString());
    }
}